Kernel/LPO.cpp

```diff
@@ -257,7 +253,7 @@ Ordering::Result LPO::majo(Term* s, TermList* tl, unsigned arity) const
   ASS(s->shared());
 
   for (unsigned i = 0; i < arity; i++) {
-    switch(lpo(s, *(tl - i))) {
```

Comment:
Same comment about the original `LPO::lpo` override here, but more importantly isn't the logic below wrong? We'd only ever go around the loop once - I think the `GREATER` case should be `continue`, not `break`.

Reply:
`lexMAE` looks wrong from that perspective as well.

Reply:
I also thought it was an issue but the `break` is for the `switch` confusingly... One issue though with `lexMAE` is that it should return `EQUAL` not `GREATER` after the loop by the original specification, but since we check for equality in `lpo` it shouldn't ever happen.